lib/web_ui/lib/src/engine/canvaskit/initialization.dart

```diff
@@ -5,8 +5,39 @@
 // @dart = 2.10
 part of engine;
 
-/// EXPERIMENTAL: Enable the Skia-based rendering backend.
-const bool experimentalUseSkia =
+/// A JavaScript entrypoint that allows developer to set rendering backend
+/// at runtime before launching the application.
+@JS('window.flutterWebRenderer')
+external String? get requestedRendererType;
+
+/// Whether to use CanvasKit as the rendering backend.
+bool get useCanvasKit =>
+    _autoDetect ? _detectRenderer() : _useSkia;
+
+/// Returns true if CanvasKit is used.
+/// Otherwise, returns false.
+bool _detectRenderer() {
+  if (requestedRendererType != null) {
+    return requestedRendererType! == 'canvaskit';
+  }
+  // If requestedRendererType is not specified, use CanvasKit for desktop and
```

Comment:
@yjbanov @ferhatb if I'm not mistaken we are setting Canvaskit as default for desktop. I am personally not sure if we are ready for this step as testing goes:
- Majority of our unit tests are build with canvaskit.
- We have coverage in many critical areas with integration tests on Desktop browsers (text editing, font loading, keyboard, platform selection, image loading, semantics (in draft), scroll (in pr), pointer events (in pr), tree shaking, platform messages). None of them runs with Canvaskit though.

Shall we prioritize running them with Canvaskit then?

Reply:
Good point. We should definitely ramp up testing in CanvasKit mode. This PR is not changing the default yet (`--web-renderer=auto` will default to CanvasKit on desktop, but `--web-renderer` itself defaults to `html` so overall the defaults are not changing), so tests can proceed on their own schedule.

In terms of test prioritization, testing things in CanvasKit mode where the choice of renderer has known effects is a good place to start from. From your list the affected areas are: font loading, image loading. Text editing is also affected but I'm not sure our existing tests will catch the distinction, so we may need new tests. Tree shaking is affected; we use devicelab for that, so it would be best to reuse the existing tests. Other things affected by renderer choice: rendering quality, platform views. There's also performance, but our benchmarks already cover both HTML and CanvasKit.

Reply:
@angjieli tl;dr there's no action items for this PR w.r.t. running existing tests in CanvasKit mode